### PR TITLE
feat(VaTreeView): add selected node class

### LINKED ISSUE #3456: VaTreeView add selected node class
Hi Team,

I have a question around highlighting (adding the background-color property) for the selected node in the Tree component.
Is this possible?

Do let me know your thoughts around it.

Thanks & Regards,
Manjari Kumari

_Originally posted by @codeonmoon in https://github.com/epicmaxco/vuestic-ui/discussions/3448_

We could add `va-tree-node--active` class, so user can stylize any state as he wants.

### PR DESCRIPTION
**Pull request by mengqiuleo**
closes #3456

add active property in treeNodeClassComputed, and control `active` with `selectedNodeComputed.value === props.node.id`,
add style for active class.

## Description

active class style is consistent with hover style

## Markup:

<details>

```vue
const treeNodeClassComputed = useBem('va-tree-node', () => ({
     disabled: !!props.node.disabled,
     active: selectedNodeComputed.value === props.node.id,
 }))
```

```scss
  &--active > &-root::before {
    border-radius: var(--va-tree-node-border-radius);
    opacity: var(--va-tree-node-interactive-bg-opacity);
  }
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Review by m0ksem — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/va-tree-view/components/VaTreeNode/VaTreeNode.vue [hunk 1]
```diff
@@ -116,6 +116,7 @@ export default defineComponent({
 
     const treeNodeClassComputed = useBem('va-tree-node', () => ({
       disabled: !!props.node.disabled,
+      active: selectedNodeComputed.value === props.node.id,
```
[the remark this answers is not in the document]
Comment (mengqiuleo) [reply packages/ui/src/components/va-tree-view/components/VaTreeNode/VaTreeNode.vue:119]: selectedNodeComputed records the id of the currently selected node.
The code for selectedNodeComputed is here. [selectedNodeComputed](https://github.com/epicmaxco/vuestic-ui/blob/develop/packages/ui/src/components/va-tree-view/hooks/useTreeView.ts#LL56C1-L56C1)
So,  I use whether the id of the current node is consistent with selectedNodeComputed. If true, it means that the node is active.

Do I need to use the `selected` property instead?


If want to use the `selected` property, code maybe looks like this.
```js
const treeNodeClassComputed = useBem('va-tree-node', () => ({
      disabled: !!props.node.disabled,
      active: !!props.node.selected,
 }))

// when this node is clicked (each node has an onNodeClick event)
const onNodeClick = () => {
    toggleSelectNode(props.node) // todo: change selected property
}
```

In toggleSelectNode, we need to set the selected state of the current node and ensure that other nodes are not selected.
This seems to be more complicated than selectedNodeComputed.

By the way, selectedNodeComputed not only saves the current selected node, but also will trigger the update-selected method in set. However, the description of update-selected in the doc is blank at present. I can create a doc issue and solve it.

[the remark this answers is not in the document]
Comment (mengqiuleo) [reply packages/ui/src/components/va-tree-view/components/VaTreeNode/VaTreeNode.vue:119]: doc: https://ui.vuestic.dev/en/ui-elements/tree-view#events

[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/ui/src/components/va-tree-view/components/VaTreeNode/VaTreeNode.vue]: I think `selectedNodeComputed` is something old... There at least might be many nodes selected, not only one. `selectedNodeComputed` stores one id, but what if there are two nodes selected? I think we need to depend on node.checked. I also can not see applied class when I select a node. I added few modifier classes to tree-node, please double-check if it work as expected.

**Review by m0ksem — APPROVED** [review]

**Comment by m0ksem** [comment]
I'll merge this PR to include in next release, we can extend it later if needed.

**Comment by codeonmoon** [comment]
@m0ksem - Could you please let us know how to use this merged feature?

**Comment by raizdev** [comment]
I see this is implemented? @m0ksem how can I use this feature? I don't see active class in the DOM for selectedNodes

**Pull request merged**